modules/text_server_adv/register_types.cpp

```diff
@@ -32,7 +32,11 @@
 
 #include "text_server_adv.h"
 
-void preregister_text_server_adv_types() {
+void initialize_text_server_adv_module(ModuleInitializationLevel p_level) {
```

Comment:
There are still references to the old functions for the GDExtension build init few lines later:

```c++
	init_obj.register_server_initializer(&preregister_text_server_adv_types);
	init_obj.register_server_terminator(&unregister_text_server_adv_types);
```

Reply:
I am not entirely sure how to fix those, maybe in a subsequent PR ?

Reply:
With the https://github.com/godotengine/godot-cpp/pull/750 and following patch, it should work:

```diff
diff --git a/modules/text_server_adv/register_types.cpp b/modules/text_server_adv/register_types.cpp
index b4fd87e1e7..6a26584506 100644
--- a/modules/text_server_adv/register_types.cpp
+++ b/modules/text_server_adv/register_types.cpp
@@ -47,7 +47,7 @@ void initialize_text_server_adv_module(ModuleInitializationLevel p_level) {
 }
 
 void uninitialize_text_server_adv_module(ModuleInitializationLevel p_level) {
-	if (p_level != MODULE_INITIALIZATION_LEVEL_SCENE) {
+	if (p_level != MODULE_INITIALIZATION_LEVEL_SERVERS) {
 		return;
 	}
 }
@@ -65,8 +65,9 @@ extern "C" {
 GDNativeBool GDN_EXPORT textserver_advanced_init(const GDNativeInterface *p_interface, const GDNativeExtensionClassLibraryPtr p_library, GDNativeInitialization *r_initialization) {
 	GDExtensionBinding::InitObject init_obj(p_interface, p_library, r_initialization);
 
-	init_obj.register_server_initializer(&preregister_text_server_adv_types);
-	init_obj.register_server_terminator(&unregister_text_server_adv_types);
+	init_obj.register_initializer(&initialize_text_server_adv_module);
+	init_obj.register_terminator(&uninitialize_text_server_adv_module);
+	init_obj.set_minimum_library_initialization_level(MODULE_INITIALIZATION_LEVEL_SERVERS);
 
 	return init_obj.init();
 }
diff --git a/modules/text_server_adv/register_types.h b/modules/text_server_adv/register_types.h
index 0028ca99db..dfe20c860c 100644
--- a/modules/text_server_adv/register_types.h
+++ b/modules/text_server_adv/register_types.h
@@ -31,9 +31,12 @@
 #ifndef TEXT_SERVER_ADV_REGISTER_TYPES_H
 #define TEXT_SERVER_ADV_REGISTER_TYPES_H
 
-#define MODULE_TEXT_SERVER_ADV_HAS_PREREGISTER
-
+#ifdef GDEXTENSION
+#include <godot_cpp/core/class_db.hpp>
+using namespace godot;
+#else
 #include "modules/register_module_types.h"
+#endif
 
 void initialize_text_server_adv_module(ModuleInitializationLevel p_level);
 void uninitialize_text_server_adv_module(ModuleInitializationLevel p_level);
diff --git a/modules/text_server_fb/register_types.cpp b/modules/text_server_fb/register_types.cpp
index d69bd996ee..fa7b87fc17 100644
--- a/modules/text_server_fb/register_types.cpp
+++ b/modules/text_server_fb/register_types.cpp
@@ -47,7 +47,7 @@ void initialize_text_server_fb_module(ModuleInitializationLevel p_level) {
 }
 
 void uninitialize_text_server_fb_module(ModuleInitializationLevel p_level) {
-	if (p_level != MODULE_INITIALIZATION_LEVEL_SCENE) {
+	if (p_level != MODULE_INITIALIZATION_LEVEL_SERVERS) {
 		return;
 	}
 }
@@ -65,8 +65,9 @@ extern "C" {
 GDNativeBool GDN_EXPORT textserver_fallback_init(const GDNativeInterface *p_interface, const GDNativeExtensionClassLibraryPtr p_library, GDNativeInitialization *r_initialization) {
 	GDExtensionBinding::InitObject init_obj(p_interface, p_library, r_initialization);
 
-	init_obj.register_server_initializer(&preregister_text_server_fb_types);
-	init_obj.register_server_terminator(&unregister_text_server_fb_types);
+	init_obj.register_initializer(&initialize_text_server_fb_module);
+	init_obj.register_terminator(&uninitialize_text_server_fb_module);
+	init_obj.set_minimum_library_initialization_level(MODULE_INITIALIZATION_LEVEL_SERVERS);
 
 	return init_obj.init();
 }
diff --git a/modules/text_server_fb/register_types.h b/modules/text_server_fb/register_types.h
index 4c804cc028..229aec2266 100644
--- a/modules/text_server_fb/register_types.h
+++ b/modules/text_server_fb/register_types.h
@@ -31,9 +31,12 @@
 #ifndef TEXT_SERVER_FB_REGISTER_TYPES_H
 #define TEXT_SERVER_FB_REGISTER_TYPES_H
 
-#define MODULE_TEXT_SERVER_FB_HAS_PREREGISTER
-
+#ifdef GDEXTENSION
+#include <godot_cpp/core/class_db.hpp>
+using namespace godot;
+#else
 #include "modules/register_module_types.h"
+#endif
 
 void initialize_text_server_fb_module(ModuleInitializationLevel p_level);
 void uninitialize_text_server_fb_module(ModuleInitializationLevel p_level);
```

Reply:
So we can wait for https://github.com/godotengine/godot-cpp/pull/750 to be merged, then @reduz amends this PR with the above diff + re-enables `godot-cpp` CI?

Reply:
Yes.

Reply:
Thanks hugely! Will apply this today.